pkg/proxy/userspace/proxysocket.go

```diff
@@ -86,17 +91,22 @@ func (tcp *tcpProxySocket) ListenPort() int {
 	return tcp.port
 }
 
-func tryConnect(service proxy.ServicePortName, srcAddr net.Addr, protocol string, proxier *Proxier) (out net.Conn, err error) {
+// Generic function type for proxying TCP and UDP request
+type proxyMethod func(in, out net.Conn, err error)
```

Comment:
it's a little weird to pass errors INTO a function - can you document why and what the function is supposed to do with it?